noir-projects/noir-contracts/contracts/test_contract/src/main.nr

```diff
@@ -377,36 +379,42 @@ contract Test {
 
     #[aztec(private)]
     fn test_shared_mutable_private_getter_for_registry_contract(
-        contract_address_to_read: AztecAddress,
         storage_slot_of_shared_mutable: Field,
         address_to_get_in_registry: AztecAddress
-    ) {
+    ) -> Field {
         // We have to derive this slot to get the location of the shared mutable inside the Map
-        let derived_slot = dep::aztec::hash::pedersen_hash(
-            [storage_slot_of_shared_mutable, address_to_get_in_registry.to_field()],
-            0
-        );
+        let derived_slot = derive_storage_slot_in_map(storage_slot_of_shared_mutable, address_to_get_in_registry);
+
         // It's a bit wonky because we need to know the delay for get_current_value_in_private to work correctly
-        let registry_private_getter: SharedMutablePrivateGetter<AztecAddress, 5> = SharedMutablePrivateGetter::new(context, contract_address_to_read, derived_slot);
+        let registry_private_getter: SharedMutablePrivateGetter<Field, 5> = SharedMutablePrivateGetter::new(context, AztecAddress::from_field(CANONICAL_KEY_REGISTRY_ADDRESS), derived_slot);
```

Comment:
I'm not fully following here, should we not be able to use the getter in here instead or computing it ourselves? Just not getting what this function is here for anymore, it is similar logic to the getter?

I guess it is only reading and not taking from oracle if zero, but could use a bit more info in the name or a comment